### PR TITLE
Return empty sitemap if not generated yet

### LINKED ISSUE #10308: Sitemaps: Have no sitemap error return valid xml
#### Steps to reproduce the issue

1. New Jetpack site, Sitemaps enabled, before the first generation time.
2. See wp_die message.

#### What Google expects
We can help prevent errors with Google, Bing, etc by returning a sitemap, even if blank.

cc: @gravityrail

### PR DESCRIPTION
**Pull request by gravityrail**
Fixes #10308

#### Changes proposed in this Pull Request:

* Return empty sitemap instead of 404 and message that sitemap isn't generated yet
* This is so that Search Console doesn't show temporary errors when waiting for sitemap to generate

#### Testing instructions:

* Delete any existing sitemap using wp shell, like this:

```bash
$ wp shell

wp> $mgr = new Jetpack_Sitemap_Manager()
=> object(Jetpack_Sitemap_Manager)#2524 (3) {
  ["librarian":"Jetpack_Sitemap_Manager":private]=>
  object(Jetpack_Sitemap_Librarian)#2525 (0) {
  }
  ["logger":"Jetpack_Sitemap_Manager":private]=>
  NULL
  ["finder":"Jetpack_Sitemap_Manager":private]=>
  object(Jetpack_Sitemap_Finder)#2526 (0) {
  }
}
wp> $mgr->callback_action_purge_data()
=> NULL
```

* go to http://yoursiteurl.com/sitemap.xml
* should show empty sitemap like this:

sitemaps-rebuild-warning

* force generation of sitemaps again:

```bash
$ wp shell
wp> $mgr = new Jetpack_Sitemap_Manager()
=> object(Jetpack_Sitemap_Manager)#2528 (3) {
  ["librarian":"Jetpack_Sitemap_Manager":private]=>
  object(Jetpack_Sitemap_Librarian)#2529 (0) {
  }
  ["logger":"Jetpack_Sitemap_Manager":private]=>
  NULL
  ["finder":"Jetpack_Sitemap_Manager":private]=>
  object(Jetpack_Sitemap_Finder)#2530 (0) {
  }
}
wp> $mgr->callback_sitemap_cron_hook()
=> NULL
```

* Now sitemap.xml should show some content:

full-sitemap

*

### CODE REVIEW

**Comment by gravityrail** [comment]
@kraftbj I'm not sure what you mean - can you mock up an example of how this would look, visually and in terms of the markup?

**Comment by gravityrail** [comment]
Oh I think I understand - you want the countdown to rendering the real markup displayed somewhere, for admins?

**Comment by jeherve** [comment]
> you want the countdown to rendering the real markup displayed somewhere, for admins?

Yep, just like in `master` since a few days ago. Here is how it looks:
https://github.com/Automattic/jetpack/pull/10275#issuecomment-429042054

**Comment by kraftbj** [comment]
Yeah, something like:
2018-10-12 at 13 35

With it looking normal, but blank, makes me think something broke.

**Comment by gravityrail** [comment]
@kraftbj @jeherve what do you think of this?

sitemaps-rebuild-warning

File: modules/sitemaps/sitemap-stylist.php [hunk 1]
```diff
@@ -192,6 +192,12 @@ public static function sitemap_index_xsl() {
 			)
 		);
 
+		if ( current_user_can( 'manage_options' ) ) {
```
Comment (gravityrail) [inline modules/sitemaps/sitemap-stylist.php]: @jeherve @kraftbj 

This is the new part, ignore all the spaces->tabs :/

File: modules/sitemaps/sitemap-stylist.php [hunk 2]
```diff
@@ -217,60 +223,65 @@ public static function sitemap_index_xsl() {
...
-        </tr>
-      </xsl:for-each>
-    </table>
-  </div>
-  <div id='footer'>
-    <p>$generated_by</p>
-  </div>
+	<div id='description'>
+		<h1>$title</h1>
+		<xsl:choose>
```
Comment (gravityrail) [inline modules/sitemaps/sitemap-stylist.php]: @jeherve @kraftbj this is the other new part

File: modules/sitemaps/sitemap-stylist.php (hunk 1 above)
Comment (kraftbj) [reply modules/sitemaps/sitemap-stylist.php]: I haven't tried to figure out why, but spinning up a new JN site with an empty sitemap isn't showing me the `$no_nodes_warning` on the sitemap or in the style.

File: modules/sitemaps/sitemap-stylist.php [hunk 3]
```diff
@@ -217,60 +223,65 @@ public static function sitemap_index_xsl() {
...
+			<xsl:when test='not(sitemap:sitemapindex/sitemap:sitemap)'>
+				<p><strong>$no_nodes_warning</strong></p>
```
Comment (kraftbj) [inline modules/sitemaps/sitemap-stylist.php]: This is throwing a warning when not defined, causing unit tests to fail.

**Review by kraftbj — CHANGES_REQUESTED** [review]

**Comment by gravityrail** [comment]
@kraftbj I wasn't able to reproduce your bug above in JN:

works-on-jn

Can you confirm that it works for you on JN and see if you can figure out why it's not on your test rig?

**Comment by kraftbj** [comment]
I'm not sure what happened that first time, but all looks good now. This is exciting.

**Review by kraftbj — APPROVED** [review]

**Review by jeherve — APPROVED** [review]
This seems to work well. Merging.

**Pull request merged**